manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -372,37 +394,144 @@ private function getDatabaseUrl(ContainerBuilder $container): string
         );
     }
 
-    private function getMailerUrl(ContainerBuilder $container): string
+    private function getMailerDsnFromMailerUrl(string $mailerUrl): string
     {
-        if ('sendmail' === $container->getParameter('mailer_transport')) {
-            return 'sendmail://localhost';
+        if (false === $parts = parse_url($mailerUrl)) {
+            throw new \InvalidArgumentException(sprintf('The MAILER_URL "%s" is not valid.', $mailerUrl));
         }
 
-        $parameters = [];
+        $options = [
+            'transport' => null,
+            'username' => null,
+            'password' => null,
+            'host' => null,
+            'port' => null,
+            'encryption' => null,
+        ];
 
-        if ($user = $container->getParameter('mailer_user')) {
-            $parameters[] = 'username='.$this->encodeUrlParameter($user);
+        $queryOptions = [];
 
-            if ($password = $container->getParameter('mailer_password')) {
-                $parameters[] = 'password='.$this->encodeUrlParameter($password);
+        if (isset($parts['scheme'])) {
+            $options['transport'] = $parts['scheme'];
+        }
+
+        if (isset($parts['user'])) {
+            $options['username'] = rawurldecode($parts['user']);
+        }
+
+        if (isset($parts['pass'])) {
+            $options['password'] = rawurldecode($parts['pass']);
+        }
+
+        if (isset($parts['host'])) {
+            $options['host'] = rawurldecode($parts['host']);
+        }
+
+        if (isset($parts['port'])) {
+            $options['port'] = $parts['port'];
+        }
+
+        if (isset($parts['query'])) {
+            parse_str($parts['query'], $query);
+
+            foreach ($query as $key => $value) {
+                if (empty($key)) {
+                    continue;
+                }
+
+                if (\in_array($key, array_keys($options), true)) {
+                    $options[$key] = $value;
+                } else {
+                    $queryOptions[$key] = $value;
+                }
             }
         }
 
+        if (empty($options['transport'])) {
+            throw new \InvalidArgumentException(sprintf('The MAILER_URL "%s" is not valid.', $mailerUrl));
+        }
+
+        if (\in_array($options['transport'], ['mail', 'sendmail'], true)) {
+            return 'sendmail+smtp://default';
+        }
+
+        if ('gmail' === $options['transport']) {
```

Comment:
Why do we need this?

Reply:
For backwards compatibility, because with Swiftmailer you were able to define the following:

```
MAILER_URL=gmail://username:password@localhost
```

https://symfony.com/doc/current/email.html#using-gmail-to-send-emails

This would automatically use `smtp.gmail.com` with SSL on port 465. Symfony mailer supports something similar, but via an additional dependency: https://symfony.com/doc/current/components/mailer.html#transport

Reply:
Ok. You should probably add a comment explaining this, so we still know in a couple of years.